apps/web/src/pages/checkout/[packSlug].tsx

```diff
@@ -0,0 +1,184 @@
+    })
+  }, [currentBid, release])
+
+  return (
+    <DefaultLayout
+      pageTitle={
+        release.type === PackType.Auction
+          ? t('common:pageTitles.Placing Bid', { name: release.title })
+          : t('common:pageTitles.Checking Out', { name: release.title })
+      }
+      panelPadding
+    >
+      <CheckoutTemplate {...paymentProps} />
+    </DefaultLayout>
+  )
+}
+
+export const getServerSideProps: GetServerSideProps<CheckoutPageProps> = async (
+  context
+) => {
+  // Verify authentication
+  const user = await getAuthenticatedUser(context)
+  if (!user) {
+    return handleUnauthenticatedRedirect(context.resolvedUrl)
+  }
+
+  // Automatically set the method to card if the feature flags aren't set
+  if (!Environment.isWireEnabled && !Environment.isCryptoEnabled) {
+    context.query.method = 'card'
+    context.query.step = 'details'
+  }
+
+  const { packs: packTemplates } = await ApiClient.instance.getPublishedPacks({
+    locale: context.locale,
+    slug: context?.params?.packSlug as string,
+  })
+
+  // If no pack templates were found, return 404
+  if (!packTemplates || packTemplates.length === 0) {
+    return {
+      notFound: true,
+    }
+  }
+
+  // If there are no remaining packs, prohibit purchase
+  const packTemplate = packTemplates[0]
+  if (!packTemplate.available) {
+    return {
+      redirect: {
+        destination: urls.release.replace(
+          ':packSlug',
+          context?.params?.packSlug as string
+        ),
+        permanent: false,
+      },
+    }
+  }
+
+  let pack: PackAuction | null = null
+  if (packTemplate.type === PackType.Auction) {
+    // If the auction has ended, only the winning bidder can purchase
+    pack = await ApiClient.instance.getAuctionPack(packTemplate.templateId)
+    const auctionEnded = packTemplate.status === PackStatus.Expired
+    const isWinningBidder = pack.activeBid?.externalId === user.externalId
+    if (
+      (auctionEnded && !isWinningBidder) ||
+      pack.ownerExternalId === user.externalId
+    ) {
+      return {
+        redirect: {
+          destination: urls.release.replace(
+            ':packSlug',
+            context?.params?.packSlug as string
+          ),
+          permanent: false,
+        },
+      }
+    }
+  } else if (packTemplate.onePackPerCustomer) {
+    // If a non-auction pack has already been purchased once and only
+    // allows one per customer, redirect to release page
+    const { total } = await ApiClient.instance.getPacksByOwnerId(
+      user.externalId,
+      { templateIds: [packTemplate.templateId] }
+    )
+    if (total > 0) {
+      return {
+        redirect: {
+          destination: urls.release.replace(
+            ':packSlug',
+            context?.params?.packSlug as string
+          ),
+          permanent: false,
+        },
+      }
+    }
+  }
+
+  // Get release based on search query
+  const address = await ApiClient.instance.createWalletAddress({
+    idempotencyKey: uuid(),
+  })
```

Comment:
Since we cannot use `serverSideProps` with a component, we have to retrieve the wallet address here. If no address is created and therefore set, the crypto option will be disabled in the methods form, and it will redirect to the methods form if the route is hit directly when no address is available.